android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -216,34 +217,78 @@ public Object getSelectionKey()
 								}
 								return null;
 							}
+
+							@Override
+							public boolean inSelectionHotspot(@NonNull MotionEvent e)
+							{
+								// Returning true allows taps to immediately select this row.
+								return true;
+							}
 						};
 					}
 					return null;
 				}
 			},
 			StorageStrategy.createLongStorage()
-		)
-			.withSelectionPredicate(SelectionPredicates.<Long>createSelectSingleAnything())
-			.build();
-		this.tracker.addObserver(new SelectionTracker.SelectionObserver()
-		{
+		);
+
+		final KrollDict properties = proxy.getProperties();
+		final boolean editing = properties.optBoolean(TiC.PROPERTY_EDITING, false);
+		final boolean allowsSelection = properties.optBoolean(TiC.PROPERTY_ALLOWS_SELECTION_DURING_EDITING, false);
+		final boolean allowsMultipleSelection
+			= properties.optBoolean(TiC.PROPERTY_ALLOWS_MULTIPLE_SELECTION_DURING_EDITING, false);
+
+		if (editing && allowsSelection) {
+			if (allowsMultipleSelection) {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectAnything())
+					.build();
+			} else {
+				this.tracker = trackerBuilder.withSelectionPredicate(SelectionPredicates.createSelectSingleAnything())
+					.build();
+			}
 
-			@Override
-			public void onSelectionChanged()
-			{
-				super.onSelectionChanged();
+			if (this.tracker != null) {
+				this.tracker.addObserver(new SelectionTracker.SelectionObserver()
+				{
+					int selectionCount = 0;
+
+					@Override
+					public void onSelectionChanged()
+					{
+						super.onSelectionChanged();
+
+						selectionCount = tracker.hasSelection() ? tracker.getSelection().size() : 0;
+						selectedItems = new ArrayList<>(selectionCount);
 
-				/*if (tracker.hasSelection()) {
-					final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
-					while (i.hasNext()) {
-						final ListItemProxy proxy = i.next();
+						if (tracker.hasSelection()) {
+							final Iterator<ListItemProxy> i = tracker.getSelection().iterator();
 
-						Log.d(TAG, "SELECTED: " + proxy.getProperties().getString(TiC.PROPERTY_TITLE));
+							while (i.hasNext()) {
+								final ListItemProxy item = i.next();
+								final KrollDict selectedItem = new KrollDict();
+								final ListSectionProxy section =
+									item.getParent() != null ? (ListSectionProxy) item.getParent() : null;
```

Comment:
If you create a section with no rows, then an empty row will appear under the section with a checkbox. If you tap on the checkbox, then the above line of code will throw an exception because `item.getParent()` can't be cast to `(ListSectionProxy)`.

You can test this with your test code by changing `createSection(10, 'A')` to `createSection(0, 'A')`.

We shouldn't show a checkbox in this case of course, but we also want to make sure the empty section header isn't selectable as well. Maybe we return out here? Or perhaps we'll have to override the [SelectionPredicate](https://developer.android.com/reference/kotlin/androidx/recyclerview/selection/SelectionTracker.SelectionPredicate) class' `canSetStateAtPosition()` and `canSetStateForKey()` methods and have them return `false` for empty section headers? I'm not sure what the best solution to this is.